File: module_video/src/main/java/com/yyxnb/module_video/fragments/VideoMainFragment.java
package com.yyxnb.module_video.fragments;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentPagerAdapter;

import com.yyxnb.arch.annotations.BindRes;
import com.yyxnb.arch.base.BaseFragment;
import com.yyxnb.module_base.weight.NoScrollViewPager;
import com.yyxnb.module_video.R;
import com.yyxnb.module_video.databinding.FragmentVideoMainBinding;

import java.util.ArrayList;
import java.util.List;

/**
 * 视频首页
 */
@BindRes
public class VideoMainFragment extends BaseFragment {

    private FragmentVideoMainBinding binding;
    private NoScrollViewPager mViewPager;

    private List<Fragment> fragments;

    @Override
    public int initLayoutResId() {
        return R.layout.fragment_video_main;
    }

    @Override
    public void initView(Bundle savedInstanceState) {
        binding = getBinding();
        mViewPager = binding.mViewPager;
    }

    @Override
    public void initViewData() {
        if (fragments == null){
            fragments = new ArrayList<>();
            fragments.add(new VideoMainBottomFragment());
            fragments.add(new VideoUserFragment());
        }
        mViewPager.setAdapter(new FragmentPagerAdapter(getChildFragmentManager()) {
            @Override
            public Fragment getItem(int i) {
                return fragments.get(i);
            }

            @Override
            public int getCount() {
                return fragments.size();
            }
        });
    }
}

File: module_video/src/main/java/com/yyxnb/module_video/fragments/VideoMainFragment.kt
package com.yyxnb.module_video.fragments


import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.yyxnb.module_video.R
import kotlinx.android.synthetic.main.fragment_video_main.*


/**
 * A simple [Fragment] subclass.
 */
class VideoMainFragment() : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                               savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_video_main, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        tvStart.setOnClickListener {
        }
    }


}
